cpu: aarch64: allow sbgemm config for matmul primitive

snadampal opened the pull request:
# Description

Please include a summary of the change. Please also include relevant motivation and context. See [contribution guidelines](https://github.com/oneapi-src/oneDNN/blob/master/CONTRIBUTING.md) for more details. If the change fixes an issue not documented in the project's Github issue tracker, please document all steps necessary to reproduce it.

This is required to support precompiled graphs where the primitive gets created with the reordered (already reordered) weight tensors, so their formats are blocked and more custom.

1. allow additional blocked layout formats. 
2. use bfloat16 fast math kernels from openxla. 

Fixes # (github issue)

# Checklist

## General

- [ ] Do all unit and benchdnn tests (`make test` and `make test_benchdnn_*`) pass locally for each commit?
ran `make test` and 
`./benchdnn --matmul --mode=P --engine=cpu --allow-enum-tags-only=0 --batch=inputs/matmul/test_matmul_ci`
- [ x] Have you formatted the code using clang-format?

## Performance improvements

- [ ] Have you submitted performance data that demonstrates performance improvements?

### New features

- [ ] Have you published an RFC for the new feature?
- [ ] Was the RFC approved?
- [ ] Have you added relevant tests?

### Bug fixes

- [ ] Have you included information on how to reproduce the issue (either in a github issue or in this PR)?
- [ ] Have you added relevant regression tests?

## [RFC](https://github.com/oneapi-src/oneDNN/tree/rfcs) PR

- [ ] Does RFC document follow the [template](https://github.com/oneapi-src/oneDNN/blob/rfcs/rfcs/template.md#onednn-design-document-rfc)?
- [ ] Have you added a link to the rendered document?

cfRod commented on src/cpu/aarch64/matmul/acl_matmul_utils.cpp:
```diff
@@ -84,11 +84,8 @@ status_t init_conf_matmul(acl_matmul_conf_t &amp, memory_desc_t &src_md,
     } else {
         auto src_tag = memory_desc_matches_one_of_tag(
                 src_md, acdb, abcd, abdc, abc, acb, ab, ba);
-        auto wei_tag = memory_desc_matches_one_of_tag(
```
how is this different from the previous if block (Lines 77 to 83) which deals with the FixedFormat case? I guess the check for plain formats is removed to support passing blocked format Ab8a via wtag?

What happens when you choose a different blocked format other than --wtag=Ab8a ? e.g --wtag=aBdc8b  or ABcd8b8a ?

snadampal replied:
Hi @cfRod , this is slightly different from the regular fixed format scenario. This is required to support precompiled graphs where the primitive gets created with the reordered (already reordered) weight tensors, so their formats are blocked and more custom. 

btw, it's not just the `Ab8a`... format, even the `ab`, ` acbd` etc are coming as blocked formats, thats why I combined the logic with the `else` part (Lined 84 to 87) instead of the format::any.

I had added a similar support to `inner_product` primitive sometime back, here is the PR, please check for more context. it was more straight forward there because everything was handled as a fixed format there
https://github.com/oneapi-src/oneDNN/pull/1768

cfRod commented on tests/benchdnn/inputs/matmul/test_matmul_ci:
```diff
@@ -211,3 +211,11 @@
 
 # fp4
 --batch=test_matmul_fp4
+
```
Can you add a test case covering the change enable f32bf16fp32 datatypes i.e with --dt=f32:bf16:f32 ? Could you also paste the verbose output before and after the change?

cfRod replied:
Approved based on https://review.mlplatform.org/c/ml/ComputeLibrary/+/13341 getting merged.

snadampal committed:
cpu: aarch64: accept reordered blocked format for matmul weight tensors

This is required to support precompiled graphs where
the primitive gets created with the reordered (already reordered)
weight tensors, so their formats are blocked and more custom.

snadampal committed:
cpu: aarch64: allow sbgemm config for matmul primitive

This is required to support precompiled graphs
where the primitive gets created with the
reordered (already reordered) weight tensors,
so their formats are blocked and more custom.
When fastmath mode is enabled and graph is pre-compiled,
the weight tensors come in reordered bfloat16 format

theComputeKid reviewed (changes requested):
Looks good to me. As Crefeda said, we will merge once the corresponding commit in ACL gets merged and is included in a tag release that we can take. Either as part of this change or separately we will need to upgrade ACL in CI to test it, when the time comes.

dzarukin reviewed (commented):
Hello. Is this PR still relevant?

snadampal commented:
yes, @dzarukin , I will ping here once the corresponding ACL change is merged.

michalowski-arm commented:
Is this change still in progress? Or should we close this PR?
